src/TestEnvironment.Docker.Containers.Mssql/MssqlContainerWaiter.cs

```diff
@@ -6,38 +6,22 @@
 
 namespace TestEnvironment.Docker.Containers.Mssql
 {
-    public class MssqlContainerWaiter : IContainerWaiter<MssqlContainer>
+    public class MssqlContainerWaiter : BaseContainerWaiter<MssqlContainer>
     {
-        private readonly ILogger _logger;
-
         public MssqlContainerWaiter(ILogger logger = null)
+            : base(logger)
         {
-            _logger = logger;
         }
 
-        public async Task<bool> Wait(MssqlContainer container, CancellationToken cancellationToken)
+        protected override async Task<bool> PerformCheck(MssqlContainer container, CancellationToken cancellationToken)
         {
-            if (container == null) new ArgumentNullException(nameof(container));
-
-            try
-            {
-                using (var connection = new SqlConnection(container.GetConnectionString()))
-                using (var command = new SqlCommand("SELECT @@VERSION", connection))
-                {
-                    await connection.OpenAsync();
-                    await command.ExecuteNonQueryAsync();
-                }
+            using var connection = new SqlConnection(container.GetConnectionString());
+            using var command = new SqlCommand("SELECT @@VERSION", connection);
+                
+            await connection.OpenAsync(cancellationToken);
+            await command.ExecuteNonQueryAsync(cancellationToken);
 
-                return true;
-            }
-            catch (Exception ex) when (ex is InvalidOperationException || ex is NotSupportedException || ex is SqlException)
```

Comment:
The main feature of such specific catches was to retry operation in case of predicable problems connected to not started container. In case if problem is not related with it - propagate it backward by call stack as soon as possible to prevent 60 seconds waiting for misconfigured container or other 'hardware' problems, not related to container waiting itself.
Probably we need to discuss it with @aharshkalep

Reply:
Sorry, my fault. Forgot to update such logic. Fixed.